pynance/dash_viz/plot_flow.py

```diff
@@ -0,0 +1,186 @@
+import base64
+import io
+
+import dash
+import flask
+
+from dash.dependencies import Input, Output, State
+import dash_core_components as dcc
+import dash_html_components as html
+import plotly.graph_objs as go
+
+from pynance.textimporter import read_csv, SupportedCsvTypes
+
+external_stylesheets = ['https://codepen.io/chriddyp/pen/bWLwgP.css']
+
+server = flask.Flask(__name__)
+app = dash.Dash(__name__,
+                external_stylesheets=external_stylesheets,
+                server=server)
+
+app.layout = html.Div([
+    html.H2("Incoming and outcoming cash"),
+    html.Table([
+        html.Tr([
+            html.Td(dcc.Dropdown(
+                id='csvtype-selection',
+                options=[{'label': 'DKB Cash', 'value': 'DKBCash'},
+                         {'label': 'DKB Visa', 'value': 'DKBVisa'}
+                         ],
+                style={'width': 200},
+                placeholder="Select csv type",
+                value=None,
+                searchable=False,
+                clearable=False)),
+            html.Td(dcc.Upload(
+                id='uploader',
+                children=html.Div([
+                    'Drag and Drop or ',
+                    html.A('Select Files')
+                ]),
+                style={
+                    'width': '300px',
+                    'height': '60px',
+                    'lineHeight': '60px',
+                    'borderWidth': '1px',
+                    'borderStyle': 'dashed',
+                    'borderRadius': '5px',
+                    'textAlign': 'center',
+                    'margin': '10px'
+                },
+                multiple=False,
+
+            ))
+        ]),
+    ]),
+    dcc.Store(id='csvtype',
+              storage_type='session'),
+
+    dcc.Graph(
+        figure=go.Figure(
+            data=[],
+        ),
+        style={'height': 500},
+        id='my-graph'
+    )
+])
+
+
+def parse_contents(contents, csvtype_str):
+    """
+    Format the undecoded content of a csv file to a dataframe
+
+    Params:
+    -------
+    contents: byte-like
+        undecoded content of a file to decode
+    csvtype_str: str
+        name of a supported csv type, should be name of the attribute of
+        SupportedCsvTypes
+
+    Returns:
+    --------
+    DataFrame
+        Data from the file, formatted as defined by CSV-Type
+    """
+    csvtype_desc = csvtype_string2description(csvtype_str)
+
+    content_string = contents.split(',')[1]

```

Comment:
What happens if `contents` does not contain a `,`?

Reply:
well, this means `contents` is ill-formatted. Changed the docstring to specify this and have general `IOError` raised if the decoding fails. `read_csv` is taken out of the `try..catch` block, because it should raise it's own exceptions

Reply:
Ok, nice!